lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -220,6 +222,15 @@ public void Deliver(byte[] message)
         public void OnNext(ICloseEvent value)
         {
             Logger.Log(Level.Info, "TaskRuntime::OnNext(ICloseEvent value)");
+            try
+            {
+                _closeHandlerFuture.Get().OnNext(value);
+            }
+            catch (Exception ex)
+            {
+                var closeEx = new TaskCloseHandlerNotBoundException("Unable to close task.", ex);
+                Utilities.Diagnostics.Exceptions.CaughtAndThrow(closeEx, Level.Error, Logger);
```

Comment:
That's right. I think I just not catch any exception here. If the handler for the task throw any exception, that would automatically go to Task fail case and the task exception will be passed back to the driver. 
I will add a test case for it.